lib/vu_state.go

```diff
@@ -93,65 +93,39 @@ type State struct {
 	GetScenarioGlobalVUIter func() uint64
 }
 
-// TagMap is a safe-concurrent Tags lookup.
-type TagMap struct {
+// VUStateTags wraps the current VU's tags and ensures a thread-safe way to
+// access and modify them exists. This is necessary because the VU tags and
+// metadata can be modified from the JS scripts via the `vu.tags` API in the
+// `k6/execution` built-in module.
+type VUStateTags struct {
 	mutex sync.RWMutex
 	tags  *metrics.TagSet
+	// TODO: Add metadata map[string]string
 }
 
-// NewTagMap creates a TagMap,
-// if a not-nil map is passed then it will be used as the internal map
-// otherwise a new one will be created.
-func NewTagMap(tags *metrics.TagSet) *TagMap {
+// NewVUStateTags initializes a new VUStateTags and returns it. It's important
+// that tags is not nil and initialized via metrics.Registry.RootTagSet().
+func NewVUStateTags(tags *metrics.TagSet) *VUStateTags {
 	if tags == nil {
-		panic("the metrics.TagSet must be initialized for creating a new lib.TagMap")
+		panic("the metrics.TagSet must be initialized for creating a new lib.VUStateTags")
 	}
-	return &TagMap{
-		tags:  tags,
+	return &VUStateTags{
 		mutex: sync.RWMutex{},
+		tags:  tags,
+		// metadata is intentionally nil by default
 	}
 }
 
-// BranchOut creates a TagSet starting from the state of the current Set.
-func (tg *TagMap) BranchOut() *metrics.TagSet {
+// GetCurrentValues returns the value of the VU tags in a thread-safe way.
+func (tg *VUStateTags) GetCurrentValues() *metrics.TagSet {
```

Comment:
Maybe just call it `TagSet`  ?

Reply:
:-1: for a couple of reasons:
- it is already obvious from the signature it will return a TagSet, so that will just duplicate it
- having "Current" in the name communicates that the value is mutable and might change